habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -470,6 +470,218 @@ def step(self, *args, is_last_action, **kwargs):
+                        shape=(2,), low=-lim, high=lim, dtype=np.float32
+                    )
+                }
+            )
+
+    def collision_check(
+        self, trans, target_trans, target_rigid_state, compute_sliding
+    ):
+        """
+        trans: the transformation of the current location of the robot
+        target_trans: the transformation of the target location of the robot given the center original Navmesh
+        target_rigid_state: the target state of the robot given the center original Navmesh
+        compute_sliding: if we want to compute sliding or not
+        """
+        num_check_cylinder = len(self._navmesh_offset)
+        # Get the offset positions
+        cur_pos = []
+        goal_pos = []
+        for i in range(num_check_cylinder):
+            cur_pos.append(
+                trans.transform_point(
+                    np.array(
+                        [
+                            self._navmesh_offset[i][0],
+                            0,
+                            self._navmesh_offset[i][1],
+                        ]
+                    )
+                )
+            )
+            goal_pos.append(
+                target_trans.transform_point(
+                    np.array(
+                        [
+                            self._navmesh_offset[i][0],
+                            0,
+                            self._navmesh_offset[i][1],
+                        ]
+                    )
+                )
+            )
+
+        # For step filter of offset positions
+        end_pos = []
+        for i in range(num_check_cylinder):
+            pos = self._sim.step_filter(cur_pos[i], goal_pos[i])
+            # Sanitize the height
+            pos[1] = 0.0
+            cur_pos[i][1] = 0.0
+            goal_pos[i][1] = 0.0
+            end_pos.append(pos)
+
+        # Planar move distance clamped by NavMesh
+        move = []
+        for i in range(num_check_cylinder):
+            move.append((end_pos[i] - goal_pos[i]).length())
+
+        # Wrap the move direction if we use sliding
+        # Find the largest diff moving direction, which means that there is a collision in that cylinder
+        if compute_sliding:
+            max_idx = np.argmax(move)
+            move_vec = end_pos[max_idx] - cur_pos[max_idx]
+            new_end_pos = trans.translation + move_vec
+            new_target_trans = mn.Matrix4.from_(
+                target_rigid_state.rotation.to_matrix(), new_end_pos
+            )
+        else:
+            new_target_trans = None
+
+        # For detection of linear or angualr velocities
+        # There is a collision if the difference between the clamped NavMesh position and target position is too great for any point.
+        diff = len([v for v in move if v > self._collision_threshold])
+
+        if diff > 0:
+            return True, new_target_trans
+        else:
+            return False, new_target_trans
```

Comment:
Instead of None, consider always returning the finalized transform for easier downstream use:
```suggestion
        
        # For detection of linear or angualr velocities
        # There is a collision if the difference between the clamped NavMesh position and target position is too great for any point.
        diff = len([v for v in move if v > self._collision_threshold])
        
        if diff > 0:
            if compute_sliding:
                max_idx = np.argmax(move)
                move_vec = end_pos[max_idx] - cur_pos[max_idx]
                new_end_pos = trans.translation + move_vec
                return True, mn.Matrix4.from_(
                    target_rigid_state.rotation.to_matrix(), new_end_pos
                )
            return True, trans
        else:
            return False, target_trans
```